minio/minio.go

```diff
@@ -19,14 +19,36 @@ import (
 )
 
 type MinioI interface {
-	UploadFile(ctx context.Context, filePath string, fileContent any, fileMimeType string) (url string, objectInfo *miniogo.ObjectInfo, err error)
-	UploadFileBytes(ctx context.Context, logger *zap.Logger, filePath string, fileBytes []byte, fileMimeType string) (url string, objectInfo *miniogo.ObjectInfo, err error)
+	UploadFile(ctx context.Context, logger *zap.Logger, param *UploadFileParam) (url string, objectInfo *miniogo.ObjectInfo, err error)
+	UploadFileBytes(ctx context.Context, logger *zap.Logger, param *UploadFileBytesParam) (url string, objectInfo *miniogo.ObjectInfo, err error)
 	DeleteFile(ctx context.Context, logger *zap.Logger, filePath string) (err error)
 	GetFile(ctx context.Context, logger *zap.Logger, filePath string) ([]byte, error)
 	GetFilesByPaths(ctx context.Context, logger *zap.Logger, filePaths []string) ([]FileContent, error)
 }
 
-const Location = "us-east-1"
+type ExpiryGroup string
+
+func (g ExpiryGroup) String() string {
+	return string(g)
+}
+
+const (
+	Location = "us-east-1"
+
+	StatusEnabled = "Enabled"
+	ExpiryTag     = "expiry-group"
+
+	FreePlanExpiry       ExpiryGroup = "free-plan-expiry"
+	ProPlanExpiry        ExpiryGroup = "pro-plan-expiry"
+	TeamPlanExpiry       ExpiryGroup = "team-plan-expiry"
+	EnterprisePlanExpiry ExpiryGroup = "enterprise-plan-expiry"
+)
+
+var expiryTimeConfig = map[ExpiryGroup]int{
+	FreePlanExpiry: 3,
+	ProPlanExpiry:  7,
+	TeamPlanExpiry: 30,
```

Comment:
This logic seems to be cloud-only, but we're defining this in a public repository. How is CE going to handle this? Is there a way to move the cloud-related logic to a private repo?

Reply:
let me check

Reply:
I'll fix this together with ur comment above